packages/adapters/signers/src/eigen/kms.rs

```diff
@@ -0,0 +1,182 @@
+use async_trait::async_trait;
+use aws_sdk_kms::{Client as AwsKmsClient, primitives::Blob};
+use k256::{
+    ecdsa::{RecoveryId, Signature, VerifyingKey},
+    pkcs8::DecodePublicKey,
+};
+use rust_eigenda_signers::{Message, RecoverableSignature};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("KMS GetPublicKey operation failed")]
+    KmsGetPublicKey(#[from] aws_sdk_kms::operation::get_public_key::GetPublicKeyError),
+
+    #[error("KMS Sign operation failed")]
+    KmsSign(#[from] aws_sdk_kms::operation::sign::SignError),
+
+    #[error("Failed to get public key from KMS")]
+    MissingPublicKey,
+
+    #[error("Failed to get signature from KMS")]
+    MissingSignature,
+
+    #[error("Failed to parse DER-encoded public key")]
+    InvalidPublicKeyDer(#[source] k256::pkcs8::spki::Error),
+
+    #[error("Failed to parse DER-encoded signature")]
+    InvalidSignatureDer(#[source] k256::ecdsa::Error),
+
+    #[error("Failed to convert public key format")]
+    PublicKeyConversion(#[source] rust_eigenda_signers::PublicKeyError),
+
+    #[error("Invalid recoverable signature")]
+    InvalidRecoverableSignature(#[source] rust_eigenda_signers::InvalidRecoveryId),
+
+    #[error("Failed to parse SEC1-encoded public key")]
+    InvalidSec1PublicKey(#[source] k256::elliptic_curve::Error),
+
+    #[error("Could not determine recovery ID for signature")]
+    RecoveryIdNotFound,
+
+    #[error("Invalid recovery ID: {0}")]
+    InvalidRecoveryId(u8),
+}
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, Clone)]
+pub struct Signer {
+    key_id: String,
+    client: AwsKmsClient,
+    public_key: rust_eigenda_signers::PublicKey,
+    k256_verifying_key: VerifyingKey,
+}
+
+impl Signer {
+    pub async fn new(client: AwsKmsClient, key_id: String) -> Result<Self> {
+        let response = client
+            .get_public_key()
+            .key_id(&key_id)
+            .send()
+            .await
+            .map_err(|e| e.into_service_error())?;
+
+        let public_key_der = response
+            .public_key
+            .ok_or(Error::MissingPublicKey)?
+            .into_inner();
+
+        // Parse the DER-encoded public key using k256
+        let k256_verifying_key = VerifyingKey::from_public_key_der(&public_key_der)
+            .map_err(Error::InvalidPublicKeyDer)?;
+
+        // Convert k256 public key to rust_eigenda_signers public key
+        // Use uncompressed format directly
+        let uncompressed_bytes = k256_verifying_key.to_encoded_point(false);
+        let public_key = rust_eigenda_signers::PublicKey::new(
+            uncompressed_bytes
+                .as_bytes()
+                .try_into()
+                .expect("uncompressed point is always 65 bytes"),
+        )
+        .map_err(Error::PublicKeyConversion)?;
```

Comment:
nit: may want to propagate errors instead of panic:

```suggestion
        let uncompressed_array: [u8; 65] = uncompressed_bytes
            .as_bytes()
            .try_into()
            .map_err(|_| Error::PublicKeyConversion(rust_eigenda_signers::PublicKeyError::InvalidLength))?;
        let public_key = rust_eigenda_signers::PublicKey::new(uncompressed_array)
            .map_err(Error::PublicKeyConversion)?;
```

Reply:
addressed in 733dcd6, needed a custom error for it